docs/i18n/20-i18n-dojo-apps.md

```diff
@@ -0,0 +1,213 @@
+
+### LocaleSwitcher Properties
+
+-   `renderer`: (updateLocale(localeData: LocaleData) => void): DNode | DNode[]
+    -   A function that is called with an `updateLocale` argument and should return `DNode | DNode[]`. The implementation's return value will be rendered, and can for example provide users a list of available locales that they can select from. In response to a user selecting a locale option, the implementation can then call `updateLocale` to enact a locale change within the whole application.
+-   `registryLabel`(optional): string
+    -   The registry label used to look up the i18n injector that will be invoked on an `updateLocale` call. This property does not need to be set when using the [registerI18nInjector](#providing-locale-data-to-i18n-aware-widgets) utility.
+
+### Example Usage
+
+The following example shows an i18n-aware widget that uses `LocaleSwitcher` to render two buttons that allow switching the application locale between English and French.
+
+```ts
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import I18ndMixin, { LocaleSwitcher, UpdateLocale } from '@dojo/framework/widget-core/mixins/I18n';
+
+import nlsBundle from '../nls/main';
+
+class MyApp extends I18ndMixin(WidgetBase) {
+	protected render() {
+		const { messages } = this.localizeBundle(nlsBundle);
+
+		return v('div', [
+			w(LocaleSwitcher, {
+				renderer: (updateLocale: UpdateLocale) => {
+					return v('div', [
+						v(
+							'button',
+							{
+								onclick: () => {
+									updateLocale({ locale: 'en' });
+								}
+							},
+							['English']
+						),
+						v(
+							'button',
+							{
+								onclick: () => {
+									updateLocale({ locale: 'fr' });
+								}
+							},
+							['French']
+						)
+					]);
+				}
+			}),
+			v('div', [messages.greetings])
+		]);
+	}
+}
+```
+
+## Overriding locales and bundles per-widget
+
+Widgets that use `I18nMixin` can have their [i18n widget properties](#i18nmixin-widget-properties) overridden when instantiated by a parent. This can be useful when rendering several widgets with different locales in a single application (that is, using multiple locales within one application), as well as to override the set of messages a third-party widget may be using and align them within the context of your application.
+
+Each i18n-aware widget can have its own locale by passing a `locale` widget property. If no locale is set, then the default locale is assumed.
```

Comment:
> If no locale is set, then the default locale is assumed.

The application locale, which can either be the default or the locale configured by i18n injector. Not sure if it's worth expanding on that here.

Reply:
I've added a new section to better describe how the default locale is determined.